modules/azure/src/main/java/org/testcontainers/containers/CosmosDBEmulatorContainer.java

```diff
@@ -0,0 +1,99 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import static org.testcontainers.containers.Constants.EMULATOR_CERTIFICATE_FILE_NAME;
+import static org.testcontainers.containers.Constants.EMULATOR_KEY;
+import static org.testcontainers.containers.Constants.KEYSTORE_FILE_NAME;
+import static org.testcontainers.containers.Constants.STORE_PASSWORD;
+import static org.testcontainers.containers.Constants.STORE_TYPE;
+import static org.testcontainers.containers.Constants.TEMP_DIRECTORY_NAME;
+import static org.testcontainers.utility.DockerImageName.parse;
+
+/**
+ * An Azure CosmosDB container
+ *
+ * Default port is 8081.
+ *
+ * @author Onur Kagan Ozcan
+ */
+public class CosmosDBEmulatorContainer extends GenericContainer<CosmosDBEmulatorContainer> {
+
+    public static final String LINUX_AZURE_COSMOS_DB_EMULATOR = "mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator";
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = parse(LINUX_AZURE_COSMOS_DB_EMULATOR);
+
+    private static final int PORT = 8081;
+
+    private Path tempDirectory;
+
+    private Boolean autoSetSystemTrustStoreParameters = true;
+
+    public CosmosDBEmulatorContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        withExposedPorts(PORT);
+        setWaitStrategy(new LogMessageWaitStrategy().withRegEx("(?s).*Started\\r\\n$"));
+    }
+
+    @Override
+    protected void configure() {
+        try {
+            this.tempDirectory = Files.createTempDirectory(TEMP_DIRECTORY_NAME);
+        } catch (Exception ex) {
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+        Path emulatorCertificatePath = tempDirectory.resolve(EMULATOR_CERTIFICATE_FILE_NAME);
+        KeyStoreUtils.downloadPemFromEmulator(getEmulatorEndpoint(), emulatorCertificatePath);
+        Path keyStorePath = tempDirectory.resolve(KEYSTORE_FILE_NAME);
+        KeyStoreUtils.importEmulatorCertificate(emulatorCertificatePath, keyStorePath);
```

Comment:
Is this code needed if `autoSetSystemTrustStoreParameters` is `false`?

Reply:
👍 Added to my patch list. I will remove this param.

I think the best is keep it simple. If we need a functionality in future to customize things, we can think at that moment :)